src/index.js

```diff
@@ -1,36 +1,107 @@
-    index: 3,
-  },
-];
-
-const populateToDoList = () => {
-  myToDoList.forEach((myToDoLists) => {
-    const li = `<li class="list-item">
-        <div class="list-div">
-            <input type="checkbox" class="check" name="" id="">
-            <p>${myToDoLists.description}</p>
+  populateToDoList = () => {
+    taskListContainer.innerHTML = '';
+    this.myToDoList.forEach((myToDoLists) => {
+      const li = `<li class="list-item">
+          <div class="list-div">
+            <div class="desc-flex">
+              <input type="checkbox" class="check" name="" id="">
+              <p class="desc">${myToDoLists.description}</p>
+            </div>
+            <form class="edit-form">
+            <input type="text" name="" id="" class="edit--input">
+            </form>
+          </div>
+          <div class="second-list-div">
+          <i class="fa-solid fa-ellipsis-vertical" data--option=${myToDoLists.index}></i>
+          <i class="fa-solid fa-trash-can" data--trash=${myToDoLists.index}></i>
         </div>
-        <i class="fa-solid fa-ellipsis-vertical"></i>
-    </li>`;
-    taskListContainer.insertAdjacentHTML('beforeend', li);
-  });
-};
+      </li>`;
+      taskListContainer.insertAdjacentHTML('beforeend', li);
+      const trashIcons = taskListContainer.querySelectorAll('.fa-trash-can');
+      const verticalIcons = taskListContainer.querySelectorAll(
+        '.fa-ellipsis-vertical',
+      );
 
-window.addEventListener('DOMContentLoaded', populateToDoList);
+      verticalIcons.forEach((icon) => {
+        icon.addEventListener('click', (e) => {
+          const { Option } = e.target.dataset;
+          const parent = e.target.parentElement.parentElement;
+          const editInput = parent.querySelector('.edit--input');
+          const descPara = parent.querySelector('.desc');
+          const editForm = parent.querySelector('.edit-form');
+          parent.classList.add('hide--desc');
+          editInput.value = descPara.textContent;
+          editInput.focus();
+          editForm.addEventListener('submit', (e) => {
+            e.preventDefault();
+            if (editInput.value) {
+              this.editTask(Option, editInput.value);
+              this.populateToDoList();
+              this.setLocalStorage(this.myToDoList);
+            }
+          });
+        });
+      });
+      trashIcons.forEach((icon) => {
+        icon.addEventListener('click', (e) => {
+          const { Trash } = e.target.dataset;
+          this.removeTask(Trash);
+          this.populateToDoList();
+          this.setLocalStorage(this.myToDoList);
+        }, { once: true });
+      });
+    });
+  };
+
+  addTask = (description, completed = false) => {
+    const index = this.myToDoList.length + 1;
+    const newTask = { description, completed, index };
+    this.myToDoList = [...this.myToDoList, newTask];
+  };
+
+  removeTask = (index) => {
+    this.myToDoList = this.myToDoList.filter((list) => list.index !== +index);
```

Comment:
- Hey, let's try to reset the indexes after making any deletions. Currently, your indexes are not getting the correct value. see the image below to fix this.

![image](https://github.com/JoyTaribagshaw/To-Do-List-Project/assets/59397159/ca7947ff-04d2-40ae-8fc7-b04eed6af6ed)

Reply:
Thank you so much for your review @zieeco , I have fixed the correct indexes.